### PR TITLE
feed inbound

### PR DESCRIPTION
**Pull request by hmrajas**

### CODE REVIEW

File: atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/ConsumeFeed.java [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+import org.apache.abdera.util.Constants;
+import org.apache.abdera.util.filter.WhiteListParseFilter;
+import org.apache.axiom.om.OMElement;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * ConsumeFeed uses to feeds from given Backend
+ */
+
```
Comment (RKathees) [inline atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/ConsumeFeed.java]: Remove the empty line

File: atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/ConsumeFeed.java [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    private static final Log log = LogFactory.getLog(ConsumeFeed.class.getName());
+    private long scanInterval;
+    private long scanIntervalDefined;
+    private long lastRanTime;
+    private String host;
+    private String feedType;
+    private Feed feed = null;
+    private java.util.Date lastUpdated;
+    RssInject rssInject;
+    FeedRegistryHandler feedRegistryHandler;
```
Comment (RKathees) [inline atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/ConsumeFeed.java]: Add the access modifier

File: atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/ConsumeFeed.java [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+
+    }
+
+    //check time interval
+    public void execute() {
+        try {
+            long currentTime = (new Date()).getTime();
+            if (((lastRanTime + scanInterval) <= currentTime)) {
+                lastRanTime = currentTime;
+                log.debug("lastRanTime " + lastRanTime);
```
Comment (RKathees) [inline atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/ConsumeFeed.java]: always have this debug logs within the isDebugEnabled() check

File: atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/ConsumeFeed.java [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+                log.debug("End : Feed Inbound EP : " + pathName);
+            }
+        } catch (Exception e) {
+            log.error(e.getMessage(), e);
+        }
+    }
+
+    //consume feeds
+    private void consume() throws IOException {
+
```
Comment (RKathees) [inline atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/ConsumeFeed.java]: empty line

File: atom_rss/atom_rss_feed_1.0.0/pom.xml [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+            <id>wso2</id>
+            <name>WSO2 internal Repository</name>
+            <url>http://dist.wso2.org/maven2/</url>
+            <releases>
+                <enabled>true</enabled>
+                <updatePolicy>never</updatePolicy>
+                <checksumPolicy>ignore</checksumPolicy>
+            </releases>
+        </repository>
+
```
Comment (Shakila) [inline atom_rss/atom_rss_feed_1.0.0/pom.xml]: Remove unwanted empty lines. Do this in all other places too.

File: atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/ConsumeFeed.java [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+                consume();
+            } else if (log.isDebugEnabled()) {
+                log.debug("Skip cycle since concurrent rate is higher than the scan interval : Feed Inbound EP " + pathName);
+            } else if (scanInterval < 999999999) {
+                scanInterval = scanInterval + 1000;
+            }
+            if (log.isDebugEnabled()) {
+                log.debug("End : Feed Inbound EP : " + pathName);
+            }
+        } catch (Exception e) {
```
Comment (Shakila) [inline atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/ConsumeFeed.java]: Avoid to use generic exception.

File: atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/ConsumeFeed.java [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+import org.apache.abdera.util.Constants;
+import org.apache.abdera.util.filter.WhiteListParseFilter;
+import org.apache.axiom.om.OMElement;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * ConsumeFeed uses to feeds from given Backend
+ */
+public class ConsumeFeed {
```
Comment (Shakila) [inline atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/ConsumeFeed.java]: Class name should should be a noun.

File: atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/ConsumeFeed.java [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+                    entry.setId(guid1.getText());
+
+                    Iterator values6 = omElement.getChildrenWithName(FeedEPConstant.FEED_LINK);
+                    OMElement link = (OMElement) values6.next();
+                    entry.setBaseUri(link.getText());
+                }
+            } else if (feedType.equalsIgnoreCase(FeedEPConstant.FEED_TYPE_ATOM)) {
+                feed = doc.getRoot();
+            }
+            try {
```
Comment (Shakila) [inline atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/ConsumeFeed.java]: Check whether the debug log is enabled or not.

File: atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/ConsumeFeed.java [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+                filter.add(Constants.AUTHOR);
+            }
+        } catch (Exception e) {
+            log.error(e.getMessage(), e);
+            return;
+        }
+
+        opts.setParseFilter(filter);
+        InputStream input;
+        try {
```
Comment (Shakila) [inline atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/ConsumeFeed.java]: Use one try - catch block in one method

File: atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/ConsumeFeed.java [hunk 10]
```diff
@@ -0,0 +1,219 @@
...
+        }
+        Document<Feed> doc;
+        try {
+            doc = parser.parse(input, "", opts);
+            if (doc.getRoot() == null) {
+                log.error("Please check host address or feed type");
+                return;
+            }
+        } catch (Exception e) {
+            log.error(e.getMessage());
```
Comment (Shakila) [inline atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/ConsumeFeed.java]: Log e with the message then only user can figure out the actual cause.

File: atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/FeedEP.java [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+        this.sequential = sequential;
+        this.coordination = true;
+        this.coordination = coordination;
+        this.injectingSeq = injectingSeq;
+        this.onErrorSeq = onErrorSeq;
+        this.synapseEnvironment = synapseEnvironment;
+        init1();
+    }
+
+//    public void destroy() {
```
Comment (Shakila) [inline atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/FeedEP.java]: If this is not needed, then delete this

File: atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/RssInject.java [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+            log.error("Error while creating message builder :: " + axisFault.getMessage(),
+                    axisFault);
+        }
+        if (builder == null) {
+            if (log.isDebugEnabled()) {
+                log.info("No message builder found for type '" + contentType +
+                        "'. Falling back to SOAP.");
+            }
+        }
+
```
Comment (Shakila) [inline atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/RssInject.java]: Remove unwanted empty lines.

File: atom_rss/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/CreateFeed.java [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+import org.apache.synapse.MessageContext;
+import org.wso2.carbon.connector.core.AbstractConnector;
+import org.wso2.carbon.connector.core.ConnectException;
+
+import java.util.Date;
+
+/**
+ * Create the Feeds
+ */
+public class CreateFeed extends AbstractConnector {
```
Comment (Shakila) [inline atom_rss/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/CreateFeed.java]: Class name should be a noun

File: atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/ConsumeFeed.java [hunk 14]
```diff
@@ -0,0 +1,219 @@
...
+                    log.debug("there Is No New Feed");
+                    return;
+                }
+            } catch (Exception e) {
+                log.error("Error in lastUpdate time checking " + e.getMessage());
+            }
+        } catch (Exception e) {
+            log.error(e.getMessage());
+        }
+        input.close();
```
Comment (RKathees) [inline atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/ConsumeFeed.java]: The input stream should be closed inside the finally block.

File: atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/ConsumeFeed.java [hunk 15]
```diff
@@ -0,0 +1,219 @@
...
+                    Iterator values3 = omElement.getChildrenWithName(FeedEPConstant.FEED_PUBDATE);
+                    OMElement Updated = (OMElement) values3.next();
+                    Date date;
+                    try {
+                        date = format.parse(Updated.getText());
+                        entry.setUpdated(date);
+                    } catch (ParseException e) {
+                        if (dateFormat != null) {
+                            format =
+                                    new SimpleDateFormat(dateFormat, Locale.ENGLISH);
```
Comment (RKathees) [inline atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/ConsumeFeed.java]: can't we use this statement in one line?

File: atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/ConsumeFeed.java [hunk 16]
```diff
@@ -0,0 +1,219 @@
...
+            return;
+        }
+
+        //convert RSS feeds as Atom
+        try {
+            if (feedType.equalsIgnoreCase(FeedEPConstant.FEED_TYPE_RSS)) {
+                Factory factory = Abdera.getNewFactory();
+                feed = factory.newFeed();
+                OMElement item = (OMElement) doc.getRoot();
+                Iterator values1 = item.getFirstElement().getChildrenWithName(FeedEPConstant.FEED_ITEM);
```
Comment (RKathees) [inline atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/ConsumeFeed.java]: Use meaningful name

File: atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/ConsumeFeed.java [hunk 17]
```diff
@@ -0,0 +1,219 @@
...
+
+                while (values1.hasNext()) {
+                    Entry entry = feed.insertEntry();
+                    OMElement omElement = (OMElement) values1.next();
+
+                    Iterator values2 = omElement.getChildrenWithName(FeedEPConstant.FEED_TITLE);
```
Comment (RKathees) [inline atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/ConsumeFeed.java]: here also use meaningful name.

File: atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/FeedEP.java [hunk 18]
```diff
@@ -0,0 +1,85 @@
...
+        if (properties.getProperty(FeedEPConstant.FEED_TIMEFORMAT) != null) {
+            this.dateFormat = properties.getProperty(FeedEPConstant.FEED_TIMEFORMAT);
+        }
+        this.sequential = sequential;
+        this.coordination = true;
+        this.coordination = coordination;
+        this.injectingSeq = injectingSeq;
+        this.onErrorSeq = onErrorSeq;
+        this.synapseEnvironment = synapseEnvironment;
+        init1();
```
Comment (RKathees) [inline atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/FeedEP.java]: Use a meaningful name

File: atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/FeedRegistryHandler.java [hunk 19]
```diff
@@ -0,0 +1,104 @@
...
+    private Registry registry;
+    private Object obj;
+
+    public FeedRegistryHandler() {
+        try {
+            registry = ServiceReferenceHolder.getInstance().getRegistry();
+        } catch (RegistryException e) {
+            log.error(e.getMessage());
+        }
+
```
Comment (RKathees) [inline atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/FeedRegistryHandler.java]: Unwanted empty lines

File: atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/FeedRegistryHandler.java [hunk 20]
```diff
@@ -0,0 +1,104 @@
...
+//
+//    public void deleteFromRegistry(String resourcePath) {
+//        try {
+//            registry.delete(resourcePath);
+//            log.debug(resourcePath + " Registry Deleted");
+//        } catch (RegistryException e) {
+//            log.error(e.getMessage());
+//        }
+//    }
+}
```
Comment (RKathees) [inline atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/FeedRegistryHandler.java]: Remove commented codes.

File: atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/FeedEP.java [hunk 21]
```diff
@@ -0,0 +1,85 @@
...
+//        }
+//        super.destroy();
+//    }
+
+    public Object poll() {
+        consume.execute();
+        return null;
+    }
+
+    private void init1() {
```
Comment (Shakila) [inline atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/FeedEP.java]: Use proper name for this initialization method

File: atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/FeedRegistryHandler.java [hunk 22]
```diff
@@ -0,0 +1,104 @@
...
+    private static final Log log = LogFactory.getLog(FeedRegistryHandler.class.getName());
+    private Resource resource;
+    private Registry registry;
+    private Object obj;
+
+    public FeedRegistryHandler() {
+        try {
+            registry = ServiceReferenceHolder.getInstance().getRegistry();
+        } catch (RegistryException e) {
+            log.error(e.getMessage());
```
Comment (Shakila) [inline atom_rss/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/FeedRegistryHandler.java]: Log e along with the message

File: atom_rss/org.wso2.carbon.connector/pom.xml [hunk 23]
```diff
@@ -0,0 +1,880 @@
...
+            <id>wso2-maven2-repository</id>
+            <name>WSO2 Maven2 Repository</name>
+            <url>scp://dist.wso2.org/home/httpd/dist.wso2.org/maven2/</url>
+        </repository>
+    </distributionManagement>
+    <dependencies>
+        <dependency>
+            <groupId>org.wso2.carbon</groupId>
+            <artifactId>org.wso2.carbon.mediation.initializer</artifactId>
+            <version>4.2.0</version>
```
Comment (Shakila) [inline atom_rss/org.wso2.carbon.connector/pom.xml]: It's better, if you define the versions in properties.

File: atom_rss/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/DeleteFeed.java [hunk 24]
```diff
@@ -0,0 +1,54 @@
...
+import org.apache.abdera.protocol.client.AbderaClient;
+import org.apache.abdera.protocol.client.ClientResponse;
+import org.apache.synapse.MessageContext;
+import org.wso2.carbon.connector.core.AbstractConnector;
+import org.wso2.carbon.connector.core.ConnectException;
+
+/**
+ * Delete the Existing feed by ID
+ */
+public class DeleteFeed extends AbstractConnector {
```
Comment (Shakila) [inline atom_rss/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/DeleteFeed.java]: Apply the comment which I put regarding class name

File: atom_rss/org.wso2.carbon.connector/src/test/resources/testng.xml [hunk 25]
```diff
@@ -0,0 +1,21 @@
...
+<suite name="ESBTestSuite" parallel="false">
+<listeners>
+<listener class-name="org.wso2.carbon.automation.core.PlatformExecutionManager" />
+<listener class-name="org.wso2.carbon.automation.core.PlatformSuiteManager" />
+<listener
+class-name="org.wso2.carbon.automation.core.PlatformAnnotationTransferManager" />
+<listener class-name="org.wso2.carbon.automation.core.PlatformTestManager" />
+<listener class-name="org.wso2.carbon.automation.core.PlatformReportManager" />
+</listeners>
+
```
Comment (Shakila) [inline atom_rss/org.wso2.carbon.connector/src/test/resources/testng.xml]: Remove unwanted empty lines

**Pull request closed without merging**